code/game/mecha/makeshift/lockermech.dm

```diff
@@ -0,0 +1,55 @@
+/obj/mecha/makeshift
+	desc = "A locker with stolen wires, struts, electronics and airlock servos crudely assembled into something that ressembles the functions of a mech."
+	name = "Locker Mech"
+	icon_state = "lockermech"
+	health = 200 //same health as ripley but without any of the armor/deflections making it very vulnerable
+	deflect_chance = 0
+	lights_power = 5
+	step_in = 5 //Slower than ripley
+	max_temperature = 5000
+	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0)
+	damage_absorption = list("brute"= 1, "fire"= 1, "bullet"= 1, "laser"= 1, "energy"= 1, "bomb"= 1)
+	max_equip = 2
+	wreckage = null
+	var/list/cargo = list()
+	var/cargo_capacity = 5
```

Comment:
The logic here seems like it would be implemented better as a `/datum/component`

Reply:
Not sure what you mean, the ripley code uses
```
var/list/cargo = new
var/cargo_capacity = 15
```
Is that it?

Reply:
Again, why isn't this a Ripley subtype?

Reply:
While this is similar to the Ripley, I don't want the mech to be able to get goliath plates or emagged, considering most stats are different than the Ripley aswell I'm not sure it should be a subtype.

Reply:
you may be able to prohibit what mech equipment it can put on it.

Reply:
you can override things like that.

If there's heavy overlap, make something a subtype and override procs or behavior where necessary, don't copy-paste.

Reply:
A component would make adding cargo to any mech trivial.

Reply:
If doing so is beyond your ken, moving all the cargo logic up to `mecha/working`, and re-implementing the makeshift mech under the `mecha/working` path, is acceptable